tools/backups2datalad.py

```diff
@@ -0,0 +1,301 @@
+__requires__ = ["boto3", "click", "dandi", "datalad", "requests"]
+
+"""
+IMPORTANT NOTE ABOUT GIT CREDENTIALS
+
+This script uses `datalad create-sibling-github` to create GitHub repositories
+that are then pushed to GitHub.  The first step requires either GitHub user
+credentials stored in the system's credentials store or else a GitHub OAuth
+token stored in the global Git config under `hub.oauthtoken`.  In addition,
+pushing to the GitHub remotes happens over SSH, so an SSH key that has been
+registered with a GitHub account is needed for the second step.
+"""
+
+from collections import deque
+from contextlib import contextmanager
+import logging
+import os
+from pathlib import Path
+import subprocess
+import sys
+from urllib.parse import urlparse, urlunparse
+
+import boto3
+from botocore import UNSIGNED
+from botocore.client import Config
+import click
+from dandi import girder
+from dandi.consts import dandiset_metadata_file
+from dandi.dandiarchive import navigate_url
+from dandi.dandiset import Dandiset
+from dandi.utils import get_instance
+from datalad.api import Dataset
+import requests
+
+log = logging.getLogger(Path(sys.argv[0]).name)
+
+
+@click.command()
+@click.option("--gh-org", help="GitHub organization to create repositories under")
+@click.option("-i", "--ignore-errors", is_flag=True)
+@click.argument("assetstore", type=click.Path(exists=True, file_okay=False))
+@click.argument("target", type=click.Path(file_okay=False))
+def main(assetstore, target, ignore_errors, gh_org):
+    logging.basicConfig(
+        format="%(asctime)s [%(levelname)-8s] %(name)s %(message)s",
+        datefmt="%Y-%m-%dT%H:%M:%S%z",
+        level=logging.INFO,
+        force=True,  # Override dandi's settings
+    )
+    DatasetInstantiator(
+        assetstore_path=Path(assetstore),
+        target_path=Path(target),
+        ignore_errors=ignore_errors,
+        gh_org=gh_org,
+    ).run()
+
+
+class DatasetInstantiator:
+    def __init__(
+        self, assetstore_path: Path, target_path: Path, ignore_errors=False, gh_org=None
+    ):
+        self.assetstore_path = assetstore_path
+        self.target_path = target_path
+        self.ignore_errors = ignore_errors
+        self.gh_org = gh_org
+        self.session = None
+        self._s3client = None
+
+    def run(self):
+        self.target_path.mkdir(parents=True, exist_ok=True)
+        with requests.Session() as self.session:
+            for did in self.get_dandiset_ids():
+                log.info("Syncing Dandiset %s", did)
+                ds = Dataset(str(self.target_path / did))
+                if not ds.is_installed():
+                    log.info("Creating Datalad dataset")
+                    ds.create(cfg_proc="text2git")
+                    ds.config.set("annex.backends", "SHA256E", where="local")
+                gitattrs = ds.pathobj / ".gitattributes"
+                gitattrs.write_text(gitattrs.read_text().replace("MD5E", "SHA256E"))
```

Comment:
A better way was "formalized" later on in https://github.com/datalad/datalad/issues/4978 avoiding this need for explicit replace:
```
import datalad
import datalad.api as dl
datalad.cfg.set('datalad.repo.backend', 'SHA256E', where='override')
ds = dl.Dataset('/tmp/dssss').create()
```

Reply:
Does this work in the most recent release of datalad?

Reply:
it should IIRC, try

Reply:
Updated.